src/openrct2/actions/RideSetAppearanceAction.h

```diff
@@ -21,7 +21,8 @@ enum class RideSetAppearanceType : uint8_t
     VehicleColourTrim,
     VehicleColourTernary,
     VehicleColourScheme,
-    EntranceStyle
+    EntranceStyle,
+    ToggleSellingItemColourRandomFlag
```

Comment:
Just looking at how this compares to the other game actions and the vast majority of boolean flags like this don't have a toggle but just a set function. This would be slightly more versatile for things like plugins that might not care what the current state the flag is in but wants to always set it to on or off.
Could you change it so this is setting the flag state based on the action _value

Reply:
Do you mean to do like this?

```C++
auto ride = get_ride(w->number);
if (ride != nullptr)
{
    auto rideSetAppearanceAction = RideSetAppearanceAction(w->number, RideSetAppearanceType::SellingItemColourIsRandom, !ride->SellingItemColourIsRandom, 0);
    GameActions::Execute(&rideSetAppearanceAction);
}
```

Reply:
Yup that looks great.